spark_log_parser/parsing_models/application_model.py

```diff
@@ -194,20 +195,25 @@ def __init__(self, eventlogpath, bucket=None, stdoutpath=None, debug=False):  #
 
                 elif event_type == "SparkListenerEnvironmentUpdate":
 
-                    curKeys = json_data["Spark Properties"].keys()
+                    spark_properties = json_data["Spark Properties"]
+                    event_keys = spark_properties.keys()
 
                     # This if is specifically for databricks logs
-                    if "spark.databricks.clusterUsageTags.sparkVersion" in curKeys:
-
+                    if "spark.databricks.clusterUsageTags.sparkVersion" in event_keys:
                         self.cloud_platform = "databricks"
-                        self.cloud_provider = json_data["Spark Properties"][
+                        self.cluster_id = spark_properties["spark.databricks.clusterUsageTags.clusterId"]
```

Comment:
For Databricks logs, grab the cluster ID from the `spark.databricks.clusterUsageTags.clusterId` field